### PR TITLE
Remove x-forwarded-host headers in admin panel and set default security options for development

### PR DESCRIPTION
**Pull request by alexandrebodin**
#### Description of what you did:

Wait for groups and repeatable to be merged before merging this branch

#### My PR is a:

- [ ] 💥 Breaking change
- [ ] 🐛 Bug fix
- [x] 💅 Enhancement
- [ ] 🚀 New feature

#### Main update on the:

- [ ] Admin
- [ ] Documentation
- [x] Framework
- [ ] Plugin

#### Manual testing done on the following databases:

- [ ] Not applicable
- [ ] MongoDB
- [ ] MySQL
- [ ] Postgres
- [ ] SQLite

### CODE REVIEW

**Comment by alexandrebodin** [comment]
Test if we add private fields to plugins models

**Comment by lauriejim** [comment]
In this PR you removed `csrf` option.
Can you also remove it from the `Settings Manager` please.
Here is the object to remove: https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-settings-manager/services/SettingsManager.js#L289

**Comment by alexandrebodin** [comment]
> LGTM!
> We will have to create a migration guide for this one.
> For `entry.toJSON()` that is no longer needed and will break applications.

Going to support it in the `sanitizeFunction` so it doesn't break compat

File: examples/getstarted/config/environments/development/security.json [hunk 1]
```diff
@@ -1,30 +1,28 @@
...
+      {
+        "img-src": "'self' http:"
+      },
+      "block-all-mixed-content"
+    ]
   },
   "p3p": {
-    "enabled": false,
+    "enabled": true,
     "value": ""
```
Comment (derrickmehaffy) [inline examples/getstarted/config/environments/development/security.json]: P3P on AdminUI states the value is required:

image

File: packages/strapi-generate-new/lib/resources/files/config/environments/development/security.json [hunk 2]
```diff
@@ -1,30 +1,28 @@
...
+    "policy": [
+      {
+        "img-src": "'self' http:"
+      },
+      "block-all-mixed-content"
+    ]
   },
   "p3p": {
-    "enabled": false,
+    "enabled": true,
```
Comment (derrickmehaffy) [inline packages/strapi-generate-new/lib/resources/files/config/environments/development/security.json]: Same here, P3P on adminUI states value is required

File: examples/getstarted/config/environments/development/security.json (hunk 1 above)
Comment (alexandrebodin) [reply examples/getstarted/config/environments/development/security.json]: OK I'll have to find a good default value then

Comment (derrickmehaffy) [reply examples/getstarted/config/environments/development/security.json]: TBH P3P is a pretty outdated security thing to begin with :stuck_out_tongue_closed_eyes:

Comment (alexandrebodin) [reply examples/getstarted/config/environments/development/security.json]: I guess I can just disable it for now then

Comment (derrickmehaffy) [reply examples/getstarted/config/environments/development/security.json]: :+1:

**Pull request merged**